src/main/scala/dxWDL/exec/TaskRunner.scala

```diff
@@ -171,6 +171,30 @@ case class TaskRunner(task: CallableTaskDefinition,
         }
     }
 
+    private def pullImage(dImg: String, retry_count: Int): Option[String] = {
```

Comment:
This is currently written as a recursive method. I suggest refactoring as a for loop. The recursive call is made inside the exception catch clause, which is more complex than you actually need.